src/liballoc/rc.rs

```diff
@@ -1702,14 +1700,18 @@ impl<T> Weak<T> {
     /// This can be used to safely get a strong reference (by calling [`upgrade`]
     /// later) or to deallocate the weak count by dropping the `Weak<T>`.
     ///
-    /// It takes ownership of one weak count. In case a [`null`] is passed, a dangling [`Weak`] is
-    /// returned.
+    /// It takes ownership of one weak count (with the exception of pointers created from [`new`],
+    /// as these don't have corresponding weak count).
     ///
     /// # Safety
     ///
-    /// The pointer must represent one valid weak count. In other words, it must point to `T` which
-    /// is or *was* managed by an [`Rc`] and the weak count of that [`Rc`] must not have reached
-    /// 0. It is allowed for the strong count to be 0.
+    /// The pointer must have originated from the [`from_raw`] (or [`as_raw`], provided there was
```

Comment:
`from_raw` doesn't have a link.

Reply:
(same for `sync`)

Reply:
Right, because it should have been `into_raw`. I'll send the fixup in a short while. Just making sure to run the tests locally this time.